applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_solver.py

```diff
@@ -217,15 +217,7 @@ def Initialize(self):
             self.Clear()
         mechanical_solution_strategy = self.get_mechanical_solution_strategy()
         mechanical_solution_strategy.SetEchoLevel(self.settings["echo_level"].GetInt())
-        if not self.is_restarted():
-            mechanical_solution_strategy.Initialize()
-        else:
-            # SetInitializePerformedFlag is not a member of SolvingStrategy but
-            # is used by ResidualBasedNewtonRaphsonStrategy.
-            try:
-                mechanical_solution_strategy.SetInitializePerformedFlag(True)
-            except AttributeError:
-                pass
```

Comment:
this is the important change
now the Elements/Conditions can check themselves if it is a restart and act accordingly